### ISSUE TITLE
Provided checkpoint file cannot be used. 

### ISSUE THREAD

**Issue by ili3p** [abhshkdz/neural-vqa#1]
Using the provided checkpoint with predict.

```
loading checkpoint from vqa_epoch23.49_0.5031.t7    
/root/torch/install/bin/luajit: /root/torch/install/share/lua/5.1/torch/File.lua:294: unknown object
stack traceback:
    [C]: in function 'error'
    /root/torch/install/share/lua/5.1/torch/File.lua:294: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:240: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:272: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    ...
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/nngraph/gmodule.lua:402: in function 'read'
    /root/torch/install/share/lua/5.1/torch/File.lua:270: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:288: in function 'readObject'
    /root/torch/install/share/lua/5.1/torch/File.lua:319: in function 'load'
    predict.lua:63: in main chunk
    [C]: in function 'dofile'
    /root/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
    [C]: at 0x00406670
```

Also it says 
`To reproduce results shown on this page or try your own image-question pairs, download the following and run predict.lua with the appropriate paths.`

What are the appropriate paths for answers_vocab.t7, questions_vocab.t7 and data.t7?

Btw training also doesn't work, i.e. the loss is nan:

```
Parameters: 6867121 
Batches: 3363   
Max iterations: 168150  
epoch 0.0029735355337496 loss nan
epoch 0.0059470710674993 loss nan   
epoch 0.0089206066012489 loss nan   
epoch 0.011894142134999 loss nan    
epoch 0.014867677668748 loss nan
```

**Comment by ili3p** [comment]
To answer my own question:

```
What are the appropriate paths for answers_vocab.t7, questions_vocab.t7 and data.t7?
```

They should be in the `data` folder.

**Comment by ili3p** [comment]
Training from scratch, i.e. generating answers_vocab.t7, questions_vocab.t7 and data.t7 <b>works</b>. But you need to install torch with 

```
TORCH_LUA_VERSION=LUA51 ./install.sh
```

Version LUA52 cannot be installed, and LUA21 (the default) cannot allocate enough memory.

**Comment by abhshkdz** [comment]
@ilija139 I see, thanks! And were you able to fix the `nan` training loss issue?

**Comment by abhshkdz** [comment]
To answer your first query, the checkpoint file is old. I added a bunch of things after that (Multi-layer LSTMs, thinner model checkpointing, etc), which is probably why it broke. :/ I'll fix it as soon as I can.

**Comment by ili3p** [comment]
@abhshkdz yes, the `nan` training loss is solved when training from scratch. 
These are the files I've just generated with the current version of the repo. If you want you can link them:

https://www.dropbox.com/s/ounll9zagxk8hp5/answers_vocab.t7?dl=0
https://www.dropbox.com/s/55hk5x2x2fh63sc/data.t7?dl=0
https://www.dropbox.com/s/ty6wm905oexk4rb/questions_vocab.t7?dl=0
https://www.dropbox.com/s/3vj6477ia5xmz2f/vqa_epoch15.00_0.4576.t7?dl=0

Anyway thanks for providing the code, it's easy to understand so it's useful as a starting point.

**Comment by abhshkdz** [comment]
@ilija139 I'll link these, thanks!
